index.html

```diff
@@ -101,212 +220,1281 @@ <h3>
+                    [=reject=] [=this=].{{UDPSocket/[[closedPromise]]}}
+                    with [=this=].{{UDPSocket/[[writable]]}}.`[[storedPromise]]`.
+                  <li>
+                    Otherwise, if [=this=].{{UDPSocket/[[readable]]}} is errored,
+                    [=reject=] [=this=].{{UDPSocket/[[closedPromise]]}}
+                    with [=this=].{{UDPSocket/[[readable]]}}.`[[storedPromise]]`.
+                  <li>
+                    Otherwise, [=resolve=] [=this=].{{UDPSocket/[[closedPromise]]}} with `undefined`.
+                </ul>
+            </ol>
+        </ol>
+
+        <section data-dfn-for="UDPMessage">
+
+          <h4><dfn>UDPMessage</dfn> dictionary</h4>
+
           <pre class="idl">
-          dictionary TCPSocketOptions : SocketOptions {
-            boolean noDelay;
-            boolean keepAlive;
-            [EnforceRange] unsigned long keepAliveDelay;
-          };
+            dictionary UDPMessage {
+              BufferSource   data;
+              DOMString      remoteAddress;
+              unsigned short remotePort;
+            };
           </pre>
-          <p>
-            The <dfn>{{TCPSocketOptions}}</dfn> dictionary inherits all attributes 
-            from {{SocketOptions}} and exposes optional TCP-specific options.
-          </p>
+
           <dl>
             <dt>
-              <dfn>noDelay</dfn> member
+              <dfn>data</dfn> member
             </dt>
             <dd>
-              Enables the `TCP_NODELAY` option,
-              disabling <a href=
-              "https://en.wikipedia.org/wiki/Nagle%27s_algorithm">Nagle's
-              algorithm</a>.
+              The user message represented as {{BufferSource}}.
+              <div class="note">
+                Note that for {{UDPSocket/[[readable]]}} the underlying type is always {{Uint8Array}}.
+              </div>
             </dd>
             <dt>
-              <dfn>keepAlive</dfn> member
+              <dfn>remoteAddress</dfn> member
             </dt>
             <dd>
-              Enables/disables TCP Keep-Alive by setting `SO_KEEPALIVE` option on the socket. 
-            </dd>
+              The remote address where the message came from.
+           </dd>
             <dt>
-              <dfn>keepAliveDelay</dfn> member
+              <dfn>remotePort</dfn> member
             </dt>
             <dd>
-              Specifies the Keep-Alive ping delay in milliseconds. Must be specified only if
-              {{TCPSocketOptions/keepAlive}} is set to `true`.
+              The remote port where the message came from.
             </dd>
+            <div class="note">
+              This spec currently only covers the case of "connected" UDP sockets
+              which are supposed to exchange packets with one address specified in the constructor.
+              For this reason {{UDPSocket/[[writable]]}} doesn't support sending packets to arbitrary {{UDPMessage/remoteAddress}}/{{UDPMessage/remotePort}}.
```

Comment:
After converting `chunk` to a `UDPMessage` throw a `TypeError` if an address or port is specified.

Reply:
Fixed!